bats test is failing in CI

<!-- Our AI triage system will automatically categorize your issue, assess severity
     and likelihood, apply labels, and route it. Just describe your issue clearly
     using the sections below as guidance. Remove or skip any that don't apply. -->

## Description

```
2026-04-10T15:41:11.7398247Z shell: /usr/bin/bash -e {0}
2026-04-10T15:41:11.7398489Z env:
2026-04-10T15:41:11.7398676Z   ASDF_DIR: /home/runner/.asdf
2026-04-10T15:41:11.7398904Z   ASDF_DATA_DIR: /home/runner/.asdf
2026-04-10T15:41:11.7399132Z ##[endgroup]
2026-04-10T15:41:11.7442351Z Cloning into 'test/test_helper/bats-assert'...
2026-04-10T15:41:12.7897270Z Cloning into 'test/test_helper/bats-support'...
2026-04-10T15:41:13.0527594Z ##[group]Run bats test/scripts/*.bats
2026-04-10T15:41:13.0527891Z [36;1mbats test/scripts/*.bats[0m
2026-04-10T15:41:13.0540619Z shell: /usr/bin/bash -e {0}
2026-04-10T15:41:13.0540858Z env:
2026-04-10T15:41:13.0541045Z   ASDF_DIR: /home/runner/.asdf
2026-04-10T15:41:13.0541268Z   ASDF_DATA_DIR: /home/runner/.asdf
2026-04-10T15:41:13.0541507Z ##[endgroup]
2026-04-10T15:41:13.2069755Z 1..11
2026-04-10T15:41:17.5928760Z ok 1 manual all builds a non-empty matrix
2026-04-10T15:41:19.3634341Z ok 2 manual single version limits matrix to that version
2026-04-10T15:41:21.1582411Z ok 3 changed-files for a single chart only includes that version
2026-04-10T15:41:21.2453126Z ok 4 non-chart changes produce an empty matrix and succeed
2026-04-10T15:41:22.0224833Z not ok 5 manual flow single value applies to all entries
2026-04-10T15:41:22.0234468Z # (from function `__assert_stream' in file test/test_helper/bats-assert/src/assert_output.bash, line 244,
2026-04-10T15:41:22.0237714Z #  from function `assert_output' in file test/test_helper/bats-assert/src/assert_output.bash, line 125,
2026-04-10T15:41:22.0238692Z #  in test file test/scripts/generate_chart_matrix.bats, line 99)
2026-04-10T15:41:22.0251892Z #   `assert_output "[\"upgrade-minor\"]"' failed
2026-04-10T15:41:22.0260523Z #
2026-04-10T15:41:22.0260986Z # -- output differs --
2026-04-10T15:41:22.0261470Z # expected : ["upgrade-minor"]
2026-04-10T15:41:22.0261980Z # actual   : []
2026-04-10T15:41:22.0262263Z # --
2026-04-10T15:41:22.0262523Z #
2026-04-10T15:41:24.3833630Z not ok 6 manual flow multiple values produce entries for each flow
2026-04-10T15:41:24.3842133Z # (from function `__assert_stream' in file test/test_helper/bats-assert/src/assert_output.bash, line 244,
2026-04-10T15:41:24.3845396Z #  from function `assert_output' in file test/test_helper/bats-assert/src/assert_output.bash, line 125,
2026-04-10T15:41:24.3855341Z #  in test file test/scripts/generate_chart_matrix.bats, line 111)
2026-04-10T15:41:24.3861205Z #   `assert_output '["install","upgrade-patch"]'' failed
2026-04-10T15:41:24.3869474Z #
2026-04-10T15:41:24.3870248Z # -- output differs --
2026-04-10T15:41:24.3870788Z # expected : ["install","upgrade-patch"]
2026-04-10T15:41:24.3871239Z # actual   : ["install"]
2026-04-10T15:41:24.3871572Z # --
2026-04-10T15:41:24.4766336Z #
2026-04-10T15:41:24.4766605Z ok 7 invalid manual flow causes failure
2026-04-10T15:41:26.1601383Z ok 8 upgrade-minor is filtered for versions <= 8.7
2026-04-10T15:41:29.0753671Z not ok 9 upgrade-patch is filtered for version == 8.9 via YAML config
2026-04-10T15:41:29.0761661Z # (from function `__assert_stream' in file test/test_helper/bats-assert/src/assert_output.bash, line 244,
2026-04-10T15:41:29.0763905Z #  from function `assert_output' in file test/test_helper/bats-assert/src/assert_output.bash, line 125,
2026-04-10T15:41:29.0766622Z #  in test file test/scripts/generate_chart_matrix.bats, line 163)
2026-04-10T15:41:29.0783944Z #   `assert_output '["install","upgrade-minor"]'' failed
2026-04-10T15:41:29.0791143Z #
2026-04-10T15:41:29.0791492Z # -- output differs --
2026-04-10T15:41:29.0791900Z # expected : ["install","upgrade-minor"]
2026-04-10T15:41:29.0792355Z # actual   : ["install"]
2026-04-10T15:41:29.0792680Z # --
2026-04-10T15:41:29.0792830Z #
2026-04-10T15:41:30.2054015Z ok 10 upgrade-patch is skipped for keycloak-original scenario (8.7)
2026-04-10T15:41:31.6989977Z ok 11 upgrade-patch is skipped for keycloak-mt even with manual flow
2026-04-10T15:41:31.7079760Z ##[error]Process completed with exit code 1.
2026-04-10T15:41:31.7141405Z Post job cleanup.
```

<!-- What is this about? What is the problem you want to solve? Describe the bug, feature request, task, or question. -->

## Expected vs Actual Behavior

<!-- If reporting a bug: What did you expect to happen, and what happened instead? -->

## Steps to Reproduce

<!-- If reporting a bug: Clear steps to reproduce the issue. -->

## Motivation / Use Case

<!-- If requesting a feature or task: Why is this needed? What problem does it solve? -->

## Acceptance Criteria

<!-- Define what "done" looks like. Use a checklist if possible. -->

- [ ] ...

## Additional Context

<!-- Any extra context, screenshots, values.yaml snippets, workarounds, logs, errors etc. -->

## References

<!-- Related issues, PRs, docs, or existing patterns. -->

## Environment

<!-- If relevant, include details to help us reproduce or understand the context. -->

- **Platform:** <!-- e.g. GKE, EKS, OpenShift, Kind -->
- **Helm CLI version:**
- **Chart version:**
- **Kubernetes version:**

--- a/.github/config/permitted-flows.yaml
+++ b/.github/config/permitted-flows.yaml
@@ -7,10 +7,6 @@ rules:
   - match: "<=8.7"
     deny:
       - upgrade-minor
-  - match: "==8.9"
-    deny:
-      - upgrade-patch
-      - upgrade-minor
   - match: "==8.10"
     deny:
       - upgrade-patch

